aws/resource_service_principal_instance_profile.go

```diff
@@ -0,0 +1,37 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+	"github.com/databrickslabs/terraform-provider-databricks/scim"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// ResourceServicePrincipalInstanceProfile binds service principal and instance profile
+func ResourceServicePrincipalInstanceProfile() *schema.Resource {
```

Comment:
```suggestion
func ResourceServicePrincipalRole() *schema.Resource {
```

role is wider than instance profile, though having similar semantics